chia/wallet/puzzles/delegated_genesis_checker.clvm

```diff
@@ -0,0 +1,27 @@
+; This is a "genesis checker" for use with cc.clvm.
+(mod (
+      PUBKEY
+      Truths
+      delta
+      lineage_check_result
+      inner_conditions
+      (
+        genesis_puzzle
+        genesis_solution
+      )
+    )
+
+    (include condition_codes.clvm)
+
+    (defun sha256tree1 (TREE)
+          (if (l TREE)
+              (sha256 2 (sha256tree1 (f TREE)) (sha256tree1 (r TREE)))
+              (sha256 1 TREE)))
+
+    (if lineage_check_result
+      inner_conditions
+      (c (list AGG_SIG_UNSAFE PUBKEY (sha256tree1 genesis_puzzle))
```

Comment:
Why `UNSAFE`? If this is intentional, a comment here would be really useful. My guess is that you delegate appropriate genesis puzzles with a one-time signature and then anyone can use them (so you'd probably delegate to another pubkey).

Reply:
That is correct, it's supposed to be graftroot but since everyone needs to use it, it needs to be an unsafe, we can add a comment about this forsure